src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -661,6 +669,43 @@ private void validateRenewalInfo(CompositePurchaseOrder compPO) {
       .thenApply(v -> compPO);
   }
 
+  private CompositePurchaseOrder validateFundDistributionTotal(CompositePurchaseOrder compPO) {
+    for (CompositePoLine cPoLine : compPO.getCompositePoLines()) {
+
+      if (cPoLine.getCost().getPoLineEstimatedPrice() != null) {
+        Double poLineEstimatedPrice = cPoLine.getCost().getPoLineEstimatedPrice();
+        String currency = cPoLine.getCost().getCurrency();
+        MonetaryAmount remainingAmount = Money.of(poLineEstimatedPrice, currency);
+
+        for (FundDistribution fundDistribution : cPoLine.getFundDistribution()) {
+          DistributionType dType = fundDistribution.getDistributionType();
+          Double value = fundDistribution.getValue();
+          MonetaryAmount amountValueMoney = Money.of(value, currency);
+          MonetaryAmount percentToAmount = null;
+
+          if (dType == DistributionType.PERCENTAGE) {
+            /**
+             * calculate remaining amount to carry forward, required if there are more fund distributions with percentage and
+             * percentToAmount = poLineEstimatedPrice * value/100;
+             */
+            MonetaryAmount poLineEstimatedPriceMoney = Money.of(poLineEstimatedPrice, currency);
+            // convert percent to amount
+            percentToAmount = poLineEstimatedPriceMoney.with(MonetaryOperators.percent(value));
+            amountValueMoney = percentToAmount;
+          }
+
+          remainingAmount = remainingAmount.subtract(amountValueMoney);
+
+          MonetaryAmount zeroMoney = Money.of(0, currency);
+          if (remainingAmount.isLessThan(zeroMoney)) {
```

Comment:
This condition rejects only those cases when the sum of distributions is greater than poLineEstimatedPrice

Reply:
There is an example how to handle fundDistibutions validation https://github.com/folio-org/mod-invoice/blob/1ca896afca3b6c26c7687578ebec95f0bf96170a/src/main/java/org/folio/rest/impl/InvoiceHelper.java#L524

Reply:
Got it but I _think_ we don't have to do all that. This one line change;(`if (!remainingAmount.isZero())`) will reject all cases (greater than, less than poLineEstimatePrice). Let me know if any case is missing.